### PR TITLE
cgroups: Make systemd cgroup parent support use specified slices

### PR DESCRIPTION
**Pull request by mrunalp**
Port of https://github.com/docker/libcontainer/pull/549

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by LK4D4** [comment]
Still LGTM

**Comment by mrunalp** [comment]
@rjnagal @vmarmol PTAL

File: libcontainer/cgroups/systemd/apply_systemd.go [hunk 1]
```diff
@@ -446,7 +446,7 @@ func (m *Manager) Set(container *configs.Config) error {
 }
 
 func getUnitName(c *configs.Cgroup) string {
-	return fmt.Sprintf("%s-%s.scope", c.Parent, c.Name)
+	return fmt.Sprintf("%s-%s.scope", c.ScopePrefix, c.Name)
```
Comment (rjnagal) [inline libcontainer/cgroups/systemd/apply_systemd.go]: do we need to check if ScopePrefix was specified?

Comment (rjnagal) [inline libcontainer/cgroups/systemd/apply_systemd.go]: I don't think "-xyz.scope" is valid. Pretty sure I saw systemd fail that name once :)

Comment (mrunalp) [inline libcontainer/cgroups/systemd/apply_systemd.go]: I think we need to clarify some of the behavior in #141. I will wait for some conclusion there :)

**Comment by selvik** [comment]
@mrunalp It'd be great if you could clarify the status of this PR, given that https://github.com/opencontainers/runc/issues/141 was recently closed. Also will --cgroup-parent continue to be supported for libcontainer? Any related issues that is tracking this? Thanks a lot!

**Comment by mrunalp** [comment]
@selvik I am going to close this for now and reopen it again later.

**Comment by selvik** [comment]
Thanks for the prompt response, @mrunalp! 
Just so we know whether we can wait for this feature or need to work around it - do you have any ideas on why it's being closed or when it'll be reopened ?

**Comment by hqhq** [comment]
@mrunalp why is this closed?
I think support cgroup parent would be helpful on systemd, although it's partial support, we should merge this and move on.

I opened a new PR #336 which almost the same as yours in case you don't have time handle this. Feel free to close that one and reopen this :)

**Pull request closed without merging**